app/models/version.rb

```diff
@@ -502,4 +503,11 @@ def record_push_event
     rubygem.record_event!(Events::RubygemEvent::VERSION_PUSHED, number: number, platform: platform, sha256: sha256_hex,
       pushed_by: pusher&.display_handle, version_gid: to_gid, actor_gid: pusher&.to_gid)
   end
+
+  def enqueue_web_hook_jobs
+    jobs = rubygem.web_hooks.or(WebHook.global).enabled
```

Comment:
What about to move this into background job already to ensure only 1 job is scheduled (which seems acceptable and predictable)?

Reply:
could do in a follow-up, but this is already how it works